### PR TITLE
Validation study intro and consent

### PR DESCRIPTION
**Pull request by the-kenneth**
Add intro and validation screens but navigation and posting to the backend is incomplete

### CODE REVIEW

File: src/features/register/gb/ValidationStudyConsentScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+                <ListItem>
+                  <CheckBox checked={this.state.reContacted} onPress={this.handleReContactedChange} />
+                  <Body style={styles.label}>
+                    <RegularText>
+                      I agree to be re-contacted in the future by Zoe Global Ltd on behalf of King’s College London
+                      researchers regarding this project (optional)
+                    </RegularText>
+                  </Body>
+                </ListItem>
+              </>
```
Comment (abubuwe) [inline src/features/register/gb/ValidationStudyConsentScreen.tsx:323]: These can be switched to use `CheckboxList` and `CheckboxItem` following #285.

File: src/features/multi-profile/SelectProfileScreen.tsx [hunk 2]
```diff
@@ -85,11 +85,15 @@ export default class SelectProfileScreen extends Component<RenderProps, State> {
...
+  async startAssessment(patientId: string, index: number) {
     try {
       const currentPatient = await userService.getCurrentPatient(patientId);
       this.setState({ isApiError: false });
-      Navigator.startAssessment(currentPatient);
+      if (isGBCountry() && index == 0 && (await userService.shouldAskForValidationStudy())) {
+        this.props.navigation.navigate('ValidationStudyIntro', { currentPatient });
+      } else {
+        Navigator.startAssessment(currentPatient);
+      }
```
Comment (MikeD-ZOE) [inline src/features/multi-profile/SelectProfileScreen.tsx:96]: Is it possible to move this navigation logic into Navigator?

Comment (the-kenneth) [reply src/features/multi-profile/SelectProfileScreen.tsx:96]: Will do

File: src/features/register/gb/ValidationStudyConsentScreen.tsx (hunk 1 above)
Comment (abubuwe) [reply src/features/register/gb/ValidationStudyConsentScreen.tsx:325]: Re-prompting on this for style consistency and to prevent future difficulties.

**Comment by the-kenneth** [comment]
@Abu I prefer if I merge my own stuff while still working on it, otherwise my local branch goes out of sync 👍

**Comment by selvach** [comment]
@the-kenneth Works like a charm - wired up to my backend! 

New checkboxes would make it look lit.

File: src/core/user/UserService.ts [hunk 3]
```diff
@@ -518,6 +519,21 @@ export default class UserService extends ApiClientBase
...
     i18n.locale = localeMap[countryCode] + '-' + UserService.userCountry;
   }
+
+  async shouldAskForValidationStudy() {
+    const response = await this.client.get<AskValidationStudy>('/study_consent/status/');
+    return response.data.should_ask_uk_validation_study;
+  }
+
+  setValidationStudyResponse(response: boolean, anonymizedData?: boolean, reContacted?: boolean) {
+    return this.client.post('/study_consent/', {
```
Comment (selvach) [inline src/core/user/UserService.ts:529]: Clean up task: Move to constants file. Does not have to be in this PR.

File: src/features/Navigation.ts [hunk 4]
```diff
@@ -94,15 +95,11 @@ class Navigator {
...
-      const newStack = state.routes;
-      while (newStack[newStack.length - 1].name != 'SelectProfile') {
-        newStack.pop();
-      }
-
-      return CommonActions.reset({
-        index: 0,
-        routes: [...newStack],
+      const profileScreen = state.routes.find((screen) => {
+        return screen.name == 'SelectProfile';
```
Comment (selvach) [inline src/features/Navigation.ts:99]: === here

File: src/features/multi-profile/SelectProfileScreen.tsx [hunk 5]
```diff
@@ -85,11 +85,11 @@ export default class SelectProfileScreen extends Component<RenderProps, State> {
     }
   }
 
-  async startAssessment(patientId: string) {
+  async profileSelected(patientId: string, index: number) {
     try {
       const currentPatient = await userService.getCurrentPatient(patientId);
       this.setState({ isApiError: false });
-      Navigator.startAssessment(currentPatient);
+      await Navigator.profileSelected(index == 0, currentPatient);
```
Comment (selvach) [inline src/features/multi-profile/SelectProfileScreen.tsx:92]: This function name made me have to think... It should probably be changed (in a clean up task).

**Comment by abubuwe** [comment]
> @the-kenneth Works like a charm - wired up to my backend!
> 
> New checkboxes would make it look lit.

Done!

**Pull request merged**